lib/three.ml

```diff
@@ -0,0 +1,96 @@
+type sack_priorities_t = int list
+[@@deriving show]
+
+module Part_one = struct
+  (** Internal constants for fast priority determination *)
+  let lower_a_code = (Char.code 'a')
+  let lower_z_code = (Char.code 'z')
+  let upper_a_code = (Char.code 'A')
+  let upper_z_code = (Char.code 'Z')
+  let lower_a_offset = lower_a_code - 1
+  let upper_a_offset = upper_a_code - 27
+
+  (** Determine priority of a given commmon item *)
+  let get_priority item =
+    let item_code = Char.code item in 
+    if item_code >= lower_a_code && item_code <= lower_z_code
+    then item_code - lower_a_offset
+    else if item_code >= upper_a_code && item_code <= upper_z_code
+    then item_code - upper_a_offset
+    else 0
+
+  (** Utility function *)
+  let explode s = List.init (String.length s) (String.get s)
+
+  (** Function to find common char between two strings *)
+  let rec project_common_item compartment_two compartment_one =
+    match compartment_one with
+    | i :: _ when (String.contains compartment_two i) -> 
+      (String.make 1 i) 
+    | i :: compartment_one' ->
+      project_common_item compartment_two compartment_one'
+    | [] -> ""
+
+  (** Determine priority of a given sack *)
+  let get_sack_priority sack =
```

Comment:
Provide explanation of basic conops of the execution of this function, no more than one paragraph.

Reply:
Consider adding examples.